snark/src/lib.rs

```diff
@@ -35,31 +35,25 @@ pub trait SNARK<F: PrimeField> {
     /// Errors encountered during setup, proving, or verification.
     type Error: 'static + ark_std::error::Error;
 
-    /// Takes in a description of a computation (specified in R1CS constraints),
-    /// and samples proving and verification keys for that circuit.
-    fn circuit_specific_setup<C: ConstraintSynthesizer<F>, R: RngCore + CryptoRng>(
-        circuit: C,
-        rng: &mut R,
-    ) -> Result<(Self::ProvingKey, Self::VerifyingKey), Self::Error>;
-
```

Comment:
I moved this algorithm to `CircuitSpecificSetupSNARK`; it doesn't really belong here.